lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1210 @@
+                collection_method = CollectionMethod.SWAB_AND_SEND
+
+        # site_reference refers to where the sample was collected
+        site_reference = create_site_reference(redcap_record_instance, collection_method, event_type)
+
+        initial_encounter_entry, initial_encounter_reference = create_encounter(
+            redcap_record_instance, patient_reference, site_reference,
+            location_resource_entries, event_type, collection_method)
+
+        # Skip the entire record if we can't create the enrollment encounter.
+        # Otherwise, just skip the record instance.
+        if not initial_encounter_entry:
+            if event_type == EventType.ENROLLMENT:
+                return None
+            else:
+                LOG.warning("Skipping record instance with insufficient information to construct a initial encounter")
+                continue
+
+        specimen_entry = None
+        specimen_observation_entry = None
+        specimen_received = (collection_method == CollectionMethod.SWAB_AND_SEND and \
+            is_complete('post_collection_data_entry_qc', redcap_record_instance)) or \
+            (collection_method == CollectionMethod.KIOSK and \
+            is_complete('kiosk_registration_4c7f', redcap_record_instance))
+
+        if specimen_received:
+            specimen_entry, specimen_reference = create_specimen(redcap_record_instance, patient_reference, collection_method)
+            specimen_observation_entry = create_specimen_observation_entry(
+                specimen_reference, patient_reference, initial_encounter_reference)
+        else:
+            LOG.info("Creating encounter for record instance without sample")
+
+        if specimen_received and not specimen_entry:
+            LOG.warning("Skipping record instance with insufficent information to construct a specimen")
+            continue
+
+        computed_questionnaire_entry = None
+        enrollment_questionnaire_entry = None
+        daily_questionnaire_entry = None
+        testing_determination_internal_questionnaire_entry = None
+
+        computed_questionnaire_entry = create_computed_questionnaire_response(
+            redcap_record_instance, patient_reference, initial_encounter_reference,
+            birthdate, datetime.strptime(initial_encounter_entry['resource']['period']['start'], '%Y-%m-%d'))
+
+        if event_type == EventType.ENROLLMENT:
+            enrollment_questionnaire_entry = create_enrollment_questionnaire_response(
+            enrollment, patient_reference, initial_encounter_reference)
+        else:
+            testing_determination_internal_questionnaire_entry = \
+                create_testing_determination_internal_questionnaire_response(
+                redcap_record_instance, patient_reference, initial_encounter_reference)
+
+            daily_questionnaire_entry = \
+                create_daily_questionnaire_response(
```

Comment:
Same here: it's not clear to me the motivation for processing these QuestionnaireResponse items separately from the others.

Reply:
I've separated the questionnaires conceptually.